internal/pkg/deploy/cloudformation/stack/env_integration_test.go

```diff
@@ -430,10 +432,37 @@ observability:
 			newObj := make(map[any]any)
 			require.NoError(t, yaml.Unmarshal([]byte(newTmpl), newObj))
 
-			delete(originalObj, "Metadata")
-			delete(newObj, "Metadata")
+			originalObj["Metadata"].(map[string]any)["Manifest"] = "any metadata because manifest could be different"
+			newObj["Metadata"].(map[string]any)["Manifest"] = "any metadata because manifest could be different"
 
-			require.Equal(t, originalObj, newObj)
+			compareStackTemplate(t, newObj, originalObj)
 		})
 	}
 }
+
+func compareStackTemplate(t *testing.T, actualObj, wantedObj map[any]any) {
+	actual, wanted := reflect.ValueOf(actualObj), reflect.ValueOf(wantedObj)
+	compareStackTemplateSection(t, reflect.ValueOf("Description"), actual, wanted)
```

Comment:
I've a very dumb question 😅 why do we need to do this?

Isn't it possible to do `require.Equal(t, wanted.MapIndex(reflect.ValueOf("Parameters")).Elem(), actual.MapIndex(reflect.ValueOf("Parameters")).Elem(), "compare Parameters")`?

What advantage do these functions bring over `require.Equal`?

Reply:
For better msg? `reflect.ValueOf("Parameters")).Elem()` still produces a `reflect.Value` type whereas `actual.MapIndex(key).Interface()` produces `any` type. Plus, it doesn't seem to work
```
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -14,3 +14,3 @@
            	            	  }),
            	            	- ptr: (unsafe.Pointer) 0xc0003b85a0,
            	            	+ ptr: (unsafe.Pointer) 0xc00019a4e0,
            	            	  flag: (reflect.flag) 0x15
            	Test:       	TestEnvStack_Template/generate_template_with_embedded_manifest_file_with_imported_certificates_and_SSL_Policy_and_empty_security_groups_rules_added_by_the_customer
```

Reply:
dumber question: the difference between this and the original comparison (comparing two `map[any]any`) is that now we know which section the discrepancy is in. Is this correct?

Reply:
that's part of it. Also we'll error out if any field is missing or redundant.